src/libraries/Microsoft.Extensions.Primitives/src/InplaceStringBuilder.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Extensions.Primitives
+{
+    [DebuggerDisplay("Value = {_value}")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    [Obsolete("This type is retained only for compatibility. The recommended alternative is string.Create<TState> (int length, TState state, System.Buffers.SpanAction<char,TState> action).", error: true)]
+    public struct InplaceStringBuilder
+    {
+        private int _offset;
+        private int _capacity;
+        private string? _value;
+
+        public InplaceStringBuilder(int capacity) : this()
+        {
+            if (capacity < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            }
+
+            _capacity = capacity;
+        }
+
+        public int Capacity
+        {
+            get => _capacity;
+            set
+            {
+                if (value < 0)
+                {
+                    ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.value);
+                }
+
+                // _offset > 0 indicates writing state
+                if (_offset > 0)
```

Comment:
>_offset [](http://example.com/codeflow?start=20&length=7)

this is interesting as it is not thread safe :-)

Reply:
This type is not thread safe.